src/v/cloud_storage/segment_meta_cstore.h

```diff
@@ -114,6 +115,15 @@ class frame {
         }
     }
 
+    const_iterator at(size_t index) const {
```

Comment:
not that we have to, but container::at will generally return a reference to a value_type.